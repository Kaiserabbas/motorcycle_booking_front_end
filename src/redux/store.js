import { configureStore } from '@reduxjs/toolkit';
import reserveReducer from './reserveSlice';
import motorcycleReducer from './motorcycleSlice';
import userReducer from './userSlice';

const store = configureStore({
  reducer: {
    reserve: reserveReducer,
    motorcycle: motorcycleReducer,
    user: userReducer,
  },
});

export default store;
