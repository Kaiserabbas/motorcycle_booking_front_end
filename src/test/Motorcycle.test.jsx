import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Motorcycle from '../components/motorcycle';
import store from '../redux/store';

describe('Motorcycle component', () => {
  const motorcycleData = {
    name: 'Test Motorcycle',
    model: 'Test Model',
    description: 'Test Description',
    imageLink: 'test-image-link.jpg',
  };

  test('renders not-found image when imageLink is not available', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Motorcycle motorcycle={motorcycleData} />
        </BrowserRouter>
      </Provider>,
    );
    expect(screen.queryByAltText('Notfound item')).toBeFalsy();
  });

  test('handles image loading and error events', () => {
    const currentScreen = render(
      <Provider store={store}>
        <BrowserRouter>
          <Motorcycle motorcycle={motorcycleData} />
        </BrowserRouter>
      </Provider>,
    );
    expect(currentScreen).toMatchSnapshot();
  });
});
