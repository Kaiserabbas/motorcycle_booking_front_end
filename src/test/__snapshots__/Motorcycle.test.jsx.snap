// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Motorcycle component handles image loading and error events 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="motorcycleItem"
      >
        <div
          class="motorcycleItemPicture"
          data-info="{\\"name\\":\\"Test Motorcycle\\",\\"model\\":\\"Test Model\\",\\"description\\":\\"Test Description\\",\\"imageLink\\":\\"test-image-link.jpg\\"}"
          role="button"
          tabindex="0"
        >
          <img
            alt="_"
            class="photoItem"
            src="test-image-link.jpg"
            srcset="Vespa Mota"
          />
        </div>
        <div
          class="motorcycleItemNameModel flexH"
        >
          <h4>
            Test Motorcycle
          </h4>
        </div>
        <hr
          class="bar"
        />
        <div
          class="motorcycleItemDescription flexH"
        >
          <p>
            Test Model
          </p>
        </div>
        <div
          class="motorcycleItemLinks flexH"
        >
          <p
            id="description"
          >
            Test Description
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="motorcycleItem"
    >
      <div
        class="motorcycleItemPicture"
        data-info="{\\"name\\":\\"Test Motorcycle\\",\\"model\\":\\"Test Model\\",\\"description\\":\\"Test Description\\",\\"imageLink\\":\\"test-image-link.jpg\\"}"
        role="button"
        tabindex="0"
      >
        <img
          alt="_"
          class="photoItem"
          src="test-image-link.jpg"
          srcset="Vespa Mota"
        />
      </div>
      <div
        class="motorcycleItemNameModel flexH"
      >
        <h4>
          Test Motorcycle
        </h4>
      </div>
      <hr
        class="bar"
      />
      <div
        class="motorcycleItemDescription flexH"
      >
        <p>
          Test Model
        </p>
      </div>
      <div
        class="motorcycleItemLinks flexH"
      >
        <p
          id="description"
        >
          Test Description
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
